src/_P042_Candle.ino

```diff
@@ -76,422 +83,440 @@ enum ColorType {
-        //                   RRGGBB  Use color in RRGGBB style (red, green blue) as HEX
-        //    <Brightness> : 0-255
-        // Samples:   CANDLE:2::100           Simple Candle with Default color and Brigthness at 100
-        //            CANDLE:5:FF0000:200     Blink with RED Color and Brigthness at 200
-        //            CANDLE:0::              Candle OFF
-        //            CANDLE:1::255           Candle ON - White and full brigthness
-
-        if (tmpString.startsWith(F("CANDLE:"))){
-          int idx1 = tmpString.indexOf(':');
-          int idx2 = tmpString.indexOf(':', idx1+1);
-          int idx3 = tmpString.indexOf(':', idx2+1);
-          int idx4 = tmpString.indexOf(':', idx3+1);
-          String val_Type = tmpString.substring(idx1+1, idx2);
-          String val_Color = tmpString.substring(idx2+1, idx3);
-          String val_Bright = tmpString.substring(idx3+1, idx4);
-
-          if (!val_Type.isEmpty()) {
-             if (val_Type.toInt() > -1 && val_Type.toInt() < 8) {
-                PCONFIG(4) = val_Type.toInt();     // Type
-                Candle_type = (SimType)PCONFIG(4);
-                #ifndef BUILD_NO_DEBUG
-                if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
-                  String log = F("CAND : CMD - Type : ");
-                  log += val_Type;
-                  addLogMove(LOG_LEVEL_DEBUG, log);
-                }
-                #endif
-             }
-          }
+    {
+      String tmpString = string;
+
+      // Test
+      // MQTT   : mosquitto_pub -d -t sensors/espeasy/ESP_Candle/cmd  -m "CANDLE_OFF"
+      // HTTP   : http://192.168.30.183/tools?cmd=CANDLE%3A5%3AFF0000%3A200
+      //          http://192.168.30.183/tools?cmd=CANDLE:4:FF0000:200
+      // SERIAL : CANDLE:4:FF0000:200<CR><LF>
+
+      // Commands
+      // CANDLE:<FlameType>:<Color>:<Brightness>
+      //    <FlameType>  : 1 Static Light, 2 Simple Candle, 3 Advanced Candle, 4 Police, 5 Blink, 6 Strobe, 7 Color Fader
+      //    <Color>      : n.def.  Use the default color
+      //                   RRGGBB  Use color in RRGGBB style (red, green blue) as HEX
+      //    <Brightness> : 0-255
+      // Samples:   CANDLE:2::100           Simple Candle with Default color and Brigthness at 100
+      //            CANDLE:5:FF0000:200     Blink with RED Color and Brigthness at 200
+      //            CANDLE:0::              Candle OFF
+      //            CANDLE:1::255           Candle ON - White and full brigthness
+
+      if (tmpString.startsWith(F("CANDLE:"))) {
+        int idx1          = tmpString.indexOf(':');
+        int idx2          = tmpString.indexOf(':', idx1 + 1);
+        int idx3          = tmpString.indexOf(':', idx2 + 1);
+        int idx4          = tmpString.indexOf(':', idx3 + 1);
+        String val_Type   = tmpString.substring(idx1 + 1, idx2);
+        String val_Color  = tmpString.substring(idx2 + 1, idx3);
+        String val_Bright = tmpString.substring(idx3 + 1, idx4);
```

Comment:
It looks like all of these can be made `const`. This helps the compiler to perhaps get rid of some more code during compile time.

Reply:
Applied with `const`, and also now a comma may be used instead of a, non-standard, colon separator, and the command is no longer case-sensitive.